src/chef_authn.erl

```diff
@@ -83,7 +83,13 @@
                           {time, erlang_time() | now} |
                           {path, http_path()} |
                           {get_header, header_fun()}.
-
+-ifdef(otp_18_plus).
```

Comment:
I think this makes sense for the `get_key_dir` macro, but for `ensure_binary`, I think a function with multiple heads would serve us just as well:

```
ensure_binary(X) when is_binary(X) ->
  X;
ensure_binary(X) when is_list(X) ->
  erlang:list_to_binary(X).
```

what do you think?